ecosystem/manager.py

```diff
@@ -303,12 +307,19 @@ def python_dev_tests(
         self, repo_url: str, tier: str = Tier.MAIN, python_version: str = "py39"
     ):
         """Runs tests against dev version of qiskit."""
+        # hack to fix tox's inability to install proper version of
+        # qiskit through github via deps configuration
+        additional_commands = [
+            "pip uninstall -y qiskit-terra",
+            "pip install git+https://github.com/Qiskit/qiskit-terra.git@main",
```

Comment:
I do the exact same in my qiskit workflow for the project template : https://github.com/mickahell/qiskit-ecosystem_template/blob/a042014fb90fad215ebba4f5d106168a41c86ba1/.github/workflows/qiskit.yml#L55-L56